### PR TITLE
refactor(cache): Deprecate namespace parameter in decorators

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

This PR deprecates usage of cache namespace for decorators, using the default one (equals the class name).

We could then remove `namespace` paramter per-datasource. Since the default namespace is different, it leads to cache reset.

The alternative is to apply class name transformation, e.g. `FooBarDatasource` to `datasource-foo-bar`, but maybe it's easier to just gradually move to new namespaces.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/cache/package/decorator.ts [hunk 1]
```diff
@@ -42,6 +44,15 @@ interface CacheParameters {
   ttlMinutes?: number;
 }
 
+function getClassName(instance: unknown): string | undefined {
+  // istanbul ignore if: should never happen
+  if (!is.object(instance)) {
+    return undefined;
+  }
+
+  return instance.constructor.name;
```
Comment (viceice) [inline lib/util/cache/package/decorator.ts:53]: would it make sense to do `kebab-case` here? would look much better 🤔

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/util/cache/package/decorator.ts (hunk 1 above)
Comment (zharinov) [reply lib/util/cache/package/decorator.ts:53]: Is it worth [introducing and maintaining the new code](https://github.com/renovatebot/renovate/pull/31060/files#diff-3744a7eafe7880efbc56d2f472cf9d5588ddeb724e8d233d10d8f78430a49bdcR3-R44), given that mostly we don't work directly with it in the underlying storage level?

Comment (viceice) [reply lib/util/cache/package/decorator.ts:53]: don't we already have lodash in our prod dependency chain which already has such a function?

Comment (zharinov) [reply lib/util/cache/package/decorator.ts:53]: I still think it's better to just use class name unchanged. As far as we don't want to reuse older cache keys, the less name mangling is better.

Comment (zharinov) [reply lib/util/cache/package/decorator.ts:53]: For example, if the underlying string converter slightly changes its behavior (e.g. for abbreviations), the dependency update will cause unanticipated cache reset. And we may spend time investigating load spike, or something like that.

File: lib/util/cache/package/decorator.ts [hunk 2]
```diff
@@ -42,6 +44,15 @@ interface CacheParameters {
   ttlMinutes?: number;
 }
 
+function getClassName(instance: unknown): string | undefined {
+  // istanbul ignore if: should never happen
+  if (!is.object(instance)) {
+    return undefined;
```
Comment (viceice) [inline lib/util/cache/package/decorator.ts:50]: we should throw if it's happening. undefined is a bad namespace.

or at least log a stack trace at debug level to see how it happens.

**Pull request closed without merging**